### PR TITLE
Handle deleting jvmchaos crds when the container holding the jvm rules has been deleted

### PR DESCRIPTION
**Pull request by peterkittreilly**
Signed-off-by: Peter Reilly <peter.reilly@netscout.com>

### What problem does this PR solve?

When a jvm rule causes the container containing the jvm is deleted, or the jvm to exit, one cannot delete the jvm rule

Using the hello example `hello.yaml`:

```yaml hello.yaml
apiVersion: v1
kind: Pod
metadata:
  name: helloworld
  namespace: hello
spec:
  containers:
    - name: helloworld
      # source code: https://github.com/WangXiangUSTC/byteman-example/tree/main/example.helloworld
      # this application will print log like this below:
      # 0. Hello World
      # 1. Hello World
      # ...
      image: xiang13225080/helloworld:v1.0
      imagePullPolicy: IfNotPresent
```

with the return rule `return.yaml`

```yaml return.yaml
apiVersion: chaos-mesh.org/v1alpha1
kind: JVMChaos
metadata:
  name: return
  namespace: hello
spec:
  action: return
  class: Main
  method: getnum
  value: '9999'
  mode: all
  selector:
    namespaces:
      - hello
```

```bash
kubectl apply -f return.yaml
kubectl delete pod -n hello helloworld
# the following never returns
kubectl delete -f return.yaml
```

With the rule `jvm-kill.yaml`

```yaml
apiVersion: chaos-mesh.org/v1alpha1
kind: JVMChaos
metadata:
  name: jvm-kill
  namespace: hello
spec:
  action: ruleData
  mode: all
  ruleData: "\nRULE modify return value\nCLASS Main\nMETHOD getnum\nAT ENTRY\nIF true\nDO killJVM(), return 9999\nENDRULE\n"
  selector:
    namespaces:
      - hello
```

```yaml
kubectl apply -f jvm-kill.yaml
# the following waits forever
kubectl delete -f jvm-kill.yaml
```

### What's changed and how it works?

In controllers/chaosimpl/jvmchaos/impl.go::Recover

if the code cannot find the container for the record, it will return
```go
v1alpha1.NotInjected, nil
```

if the code cannot  delete the jvm rule, it will also return

```go
v1alpha1.NotInjected, nil
```

to say to the caller that there is no need to remove the jvm rule.

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

```bash
kubectl apply -f return.yaml
kubectl delete pod -n hello helloworld
# the following now works
kubectl delete -f return.yaml
```

```bash
kubectl apply -f jvm-kill.yaml
# the following now works
kubectl delete -f jvm-kill.yaml
```

Side effects

- [ ] Breaking backward compatibility

### CODE REVIEW

File: controllers/chaosimpl/jvmchaos/impl.go
Comment (WangXiangUSTC) [inline controllers/chaosimpl/jvmchaos/impl.go:128]: How about checking the error message? Only when the error message contains something means contained not found return `v1alpha1.NotInjected`

File: controllers/chaosimpl/jvmchaos/impl.go
Comment (WangXiangUSTC) [inline controllers/chaosimpl/jvmchaos/impl.go:146]: ditto

**Review by WangXiangUSTC — COMMENTED** [review]
rest LGTM, thanks for your contribution!

File: controllers/chaosimpl/jvmchaos/impl.go
Comment (peterkittreilly) [reply controllers/chaosimpl/jvmchaos/impl.go:150]: Here I can only see the exit status of 1.

File: controllers/chaosimpl/jvmchaos/impl.go
Comment (WangXiangUSTC) [reply controllers/chaosimpl/jvmchaos/impl.go:129]: Great! Maybe it's better to put the check under L131.

File: controllers/chaosimpl/jvmchaos/impl.go
Comment (WangXiangUSTC) [reply controllers/chaosimpl/jvmchaos/impl.go:150]: OK, I will update the error next pr. How about reverting this update?

Comment (WangXiangUSTC) [reply controllers/chaosimpl/jvmchaos/impl.go:150]: I file another pr https://github.com/chaos-mesh/chaos-mesh/pull/3004, after this pr merged, the error will contain the real failed reason. How about update this pr after my pr is merged?

Comment (peterkittreilly) [reply controllers/chaosimpl/jvmchaos/impl.go:150]: Sure,  will do

Comment (peterkittreilly) [reply controllers/chaosimpl/jvmchaos/impl.go:150]: I have added the check for the real failed reason (Connection refused)

Comment (WangXiangUSTC) [reply controllers/chaosimpl/jvmchaos/impl.go:150]: Great！

**Review by WangXiangUSTC — APPROVED** [review]
LGTM, thanks for your contribution!

**Comment by WangXiangUSTC** [comment]
@chaos-mesh/committers @chaos-mesh/maintainers PTAL

**Review by YangKeao — APPROVED** [review]
LGTM

**Comment by peterkittreilly** [comment]
Thanks!
@WangXiangUSTC  and @YangKeao

**Pull request merged**
